cmd/run.go

```diff
@@ -201,9 +201,15 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) (err error) {
 					IsStdOutTTY: c.gs.Stdout.IsTTY,
 					IsStdErrTTY: c.gs.Stderr.IsTTY,
 				},
-			})
+			}
+			summaryResult, hsErr := test.initRunner.HandleSummary(globalCtx, summary)
 			if hsErr == nil {
 				hsErr = handleSummaryResult(c.gs.FS, c.gs.Stdout, c.gs.Stderr, summaryResult)
+				waitForSummaryGeneratedEvent := emitEvent(&event.Event{
+					Type: event.TestSummaryGenerated,
+					Data: summary,
```

Comment:
It should be better to copy ObservedMetrics and RootGroup instead of directly passing them. They are pointers and they might generate data races if there is a read/write across extensions and/or from k6core.